maven-core/src/main/java/org/apache/maven/internal/aether/DefaultRepositorySystemSessionFactory.java

```diff
@@ -261,9 +261,37 @@ else if ( request.isUpdateSnapshots() )
         mavenRepositorySystem.injectProxy( session, request.getPluginArtifactRepositories() );
         mavenRepositorySystem.injectAuthentication( session, request.getPluginArtifactRepositories() );
 
-        if ( Features.buildConsumer( request.getUserProperties() ).isActive() )
+        if ( !installRequestTransformers.isEmpty() )
+        {
+            if ( logger.isDebugEnabled() )
+            {
+                logger.debug( "Applying install request transformers (shown in order):" );
+                for ( InstallRequestTransformer transformer : installRequestTransformers )
+                {
+                    logger.debug( " * {}", transformer.getClass() );
+                }
+                logger.debug( "" );
+            }
+            session.getData().set(
+                    InstallRequestTransformer.KEY,
+                    new ChainedInstallRequestTransformer( installRequestTransformers )
+            );
+        }
+        if ( !deployRequestTransformers.isEmpty() )
         {
-            session.setFileTransformerManager( a -> getTransformersForArtifact( a, session.getData() ) );
```

Comment:
do we remove both `setFileTransformerManager` and `getFileTransformerManager` from `session`?
if not set what return get method?

Reply:
Do not use FileTransformer API anymore, in short. If this get's accepted, you will want to implement DeployRequestTransformer component... but let's go baby steps first, and make sure this would work for you too.